src/main/java/nextstep/subway/path/application/PathService.java

```diff
@@ -0,0 +1,51 @@
+package nextstep.subway.path.application;
+
+import nextstep.subway.line.application.LineService;
+import nextstep.subway.line.domain.Section;
+import nextstep.subway.path.domain.PathFinder;
+import nextstep.subway.path.dto.PathResponse;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.dto.StationResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class PathService {
+
+    private LineService lineService;
+    private StationService stationService;
+
+    public PathService(LineService lineService, StationService stationService) {
+        this.lineService = lineService;
+        this.stationService = stationService;
+    }
+
+    public PathResponse findShortestPath(Long sourceId, Long targetId) {
+        Station source = stationService.findById(sourceId);
+        Station target = stationService.findById(targetId);
+
+        List<Section> sections = lineService.findLines().stream()
+                .map(it -> it.getSections())
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+
+        PathFinder pathFinder = PathFinder.create(sections, source, target);
+        pathFinder.getShortestPath();
+
+        return createPathResponse(pathFinder.getShortestPath(), pathFinder.getShortestPathDistance());
+    }
```

Comment:
```suggestion
    public PathResponse findShortestPath(Long sourceId, Long targetId) {
        Station source = stationService.findById(sourceId);
        Station target = stationService.findById(targetId);

        SubwayGraph subwayGraph = graphService.findGraph();

        return createPathResponse(subwayGraph, source, target);
    }
```

이런 느낌으로 Graph를 재사용(캐싱)할 수 있도록 분리한 뒤
source와 target에 따라서 경로를 다르게 응답할 수 있게 하면 어떨까 해서 드린 피드백이었습니다 :)
이 리팩터링을 할 때 TDD를 통한 점진적인 리팩터링을 연습해볼 수 있을 것 같아요!

Reply:
넵 올려주신 부분 다시 고민해보도록 하겠습니다!

Reply:
의도 해주신 방향으로 구현했는지 잘 모르겠습니다만...
레거시 TDD연습은 열심히 한것 같네요. ㅎ -0-;

Reply:
의도 해주신 방향으로 구현했는지 잘 모르겠습니다만...
레거시 TDD연습은 열심히 한것 같네요. ㅎ -0-;